tests/api/ClientTest.php

```diff
@@ -0,0 +1,282 @@
+{
+    public function testCreateClient(): void
+    {
+        $httpClient = $this->authenticate();
+        $client = ClientMother::base();
+        $client = $this->postClient($httpClient, $client);
+        $this->assertResponseIsSuccessful();
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertJsonContains($client->getData());
+        $this->assertJsonContains($client->getContext());
+    }
+
+    public function testCreateClientAllParameters(): void 
+    {
+        $httpClient = $this->authenticate();
+        $preScript = $this->getScriptId($httpClient, 'script_client_pre');
+        $postScript = $this->getScriptId($httpClient, 'script_client_post');
+        $client = ClientMother::withAllParameters(
+            1, 
+            "some description", 
+            false, 
+            2, 
+            [$postScript], 
+            [$preScript],
+            -1, 
+            "rsync long arguments", 
+            "rsync short arguments", 
+            "ssh arguments", 
+            "root@172.0.0.1"
+        );
+        $client = $this->postClient($httpClient, $client);
+        $this->assertResponseIsSuccessful();
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertJsonContains($client->getData());
+        $this->assertJsonContains($client->getContext());
+    }
+    public function testCreateClientInvalidMaxParallelJobs(): void
+    {
+        $httpClient = $this->authenticate();
+        $client = ClientMother::withInvalidMaxParallelJobs();
+        $client = $this->postClient($httpClient, $client);
+        $this->assertResponseStatusCodeSame(422);
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertJsonContains([
+            '@context' => '/api/contexts/ConstraintViolationList',
+            '@type' => 'ConstraintViolationList',
+            'hydra:title' => 'An error occurred',
+            'hydra:description' => 'maxParallelJobs: Max parallel jobs value must be a positive integer',
+        ]);
+    }
+
+    public function testCreateClientRepeatedName(): void
+    {
+        $httpClient = $this->authenticate();
+        $client = ClientMother::base();
+        $this->postClient($httpClient, $client);
+        $this->postClient($httpClient, $client);
+        $this->assertResponseStatusCodeSame(400);
+        $this->assertHydraError("An exception occurred while executing 'INSERT INTO Client (description, isActive, name, url, quota, sshArgs, rsyncShortArgs, rsyncLongArgs, state, maxParallelJobs, data, owner_id) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)' with params [null, 1, \"".$client->getName()."\", \"\", -1, null, null, null, \"NOT READY\", 1, null, 1]:\n\nSQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '".$client->getName()."' for key 'Client.UNIQ_C0E801635E237E06'");
```

Comment:
Is such a great deal to add a comment pointing to the already opened issue?
I think is the third review where I ask for it.

Reply:
I'm sorry, I thought you meant to add the comment in the PR conversation, not inside the code. I'm on it.